scripts/commands/checkLinks.ts

```diff
@@ -119,61 +82,22 @@ async function processDocsFiles(
   filePathsToLoad: string[],
   filePathsToCheck: string[],
 ): Promise<[File[], Link[], Link[]]> {
-  const fileList: File[] = [];
+  const files: File[] = [];
   for (let filePath of filePathsToLoad) {
     const [_, anchors] = await getMarkdownAndAnchors(filePath);
-    fileList.push(new File(filePath, anchors, false));
+    files.push(new File(filePath, anchors, false));
   }
 
-  const internalLinkList: Link[] = [];
-  const externalLinkList: Link[] = [];
-  // This is a map to avoid link duplications.
-  const linkMap = new Map<string, string[]>();
-
-  for (let filePath of filePathsToCheck) {
+  const linksToOriginFiles = new Map<string, string[]>();
+  for (const filePath of filePathsToCheck) {
     const [markdown, anchors] = await getMarkdownAndAnchors(filePath);
-    fileList.push(new File(filePath, anchors, false));
-    unified()
-      .use(rehypeParse)
-      .use(remarkGfm)
-      .use(rehypeRemark)
-      .use(() => {
-        return function transform(tree: Root) {
-          visit(tree, "text", (TreeNode) => {
-            markdownLinkExtractor(String(TreeNode.value)).links.map(
-              (url: string) => {
-                let link = linkMap.get(url);
-                if (link != null) {
-                  link.push(filePath);
-                } else {
-                  linkMap.set(url, [filePath]);
-                }
-              },
-            );
-          });
-          visit(tree, "link", (TreeNode) => {
-            let link = linkMap.get(TreeNode.url);
-            if (link != null) {
-              link.push(filePath);
-            } else {
-              linkMap.set(TreeNode.url, [filePath]);
-            }
-          });
-          visit(tree, "image", (TreeNode) => {
-            let link = linkMap.get(TreeNode.url);
-            if (link != null) {
-              link.push(filePath);
-            } else {
-              linkMap.set(TreeNode.url, [filePath]);
-            }
-          });
-        };
-      })
-      .use(remarkStringify)
-      .process(markdown);
+    files.push(new File(filePath, anchors, false));
+    await addLinksToMap(filePath, markdown, linksToOriginFiles);
```

Comment:
Note that we should have been calling `await` before. I don't think it actually broke things though, fortunately.